### PR TITLE
[BD-46] feat: added Husky  configurations

### PR DESCRIPTION
**Pull request by PKulkoRaccoonGang**
## Description

- added Husky configuration

**Issue:** https://github.com/openedx/paragon/issues/2864

## Merge Checklist

* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [ ] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [ ] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [ ] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [ ] Were your changes tested in the `example` app?
* [ ] Is there adequate test coverage for your changes?
* [ ] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @PKulkoRaccoonGang!


When this pull request is ready, tag your edX technical lead.

File: README.md [hunk 1]
```diff
@@ -148,7 +148,11 @@ module.exports = {
   dist: The sub-directory of the source code where it puts its build artifact. Often "dist".
   */
   localModules: [
-    { moduleName: '@edx/paragon/scss/core', dir: '../src/paragon', dist: 'scss/core' },
+    {
+      moduleName: '@edx/paragon/scss/core',
+      dir: '../src/paragon',
+      dist: 'scss/core',
+    },
```
Comment (viktorrusakov) [inline README.md:155]: Why only the first element in the list got updated? Also, I think we shouldn't make any changes here, it was fine before when every entry is in one linel

File: README.md [hunk 2]
```diff
@@ -163,80 +167,82 @@ Then, when importing Paragon's core SCSS in your MFE the import needs to begin w
...
-          
-          export default MyFunctionComponent;
-          ```
-      
+
+   - For places where you need to display a string, and it's okay if it is a React element use `FormattedMessage`, e.g. (see [Alert](src/Alert/index.jsx) component for a full example)
+
+     ```javascript
+     import { FormattedMessage } from 'react-intl';
+     <FormattedMessage
```
Comment (viktorrusakov) [inline README.md:178]: We do need an empty line after import here

File: README.md [hunk 3]
```diff
@@ -163,80 +167,82 @@ Then, when importing Paragon's core SCSS in your MFE the import needs to begin w
...
+       class MyClassComponent extends React.Component {
+         render() {
+           const { altText, intl } = this.props;
+           const intlAltText =
+             altText ||
+             intl.formatMessage({
+               id: 'pgn.MyComponent.altText',
+               defaultMessage: 'Close',
+               description: 'Close label for Toast component',
+             });
```
Comment (viktorrusakov) [inline README.md:201]: This definitely looks wrong, `altText` shouldn't start on the next line (see line 223 in this file below, for some reason the style is different there)

File: README.md [hunk 4]
```diff
@@ -340,11 +346,8 @@ Describe your component here and give usage examples.
 ### Basic Usage
 
 ```jsx live
-<MyComponent>
-  Hello!
-</MyComponent>
+<MyComponent>Hello!</MyComponent>
```
Comment (viktorrusakov) [inline README.md:349]: Cand we not change this? The way it was before looks more readable to me

File: component-generator/templates/README.md [hunk 5]
```diff
@@ -15,7 +15,5 @@ Describe your component here and give usage examples.
 ### Basic Usage
 
 ```jsx live
-<componentName>
-  Hello!
-</componentName>
+<componentName>Hello!</componentName>
```
Comment (viktorrusakov) [inline component-generator/templates/README.md:18]: Cand we not change this? The way it was before looks more readable to me

File: src/Alert/README.md [hunk 6]
```diff
@@ -27,47 +27,50 @@ This component utilizes and extends the `Alert` component from react-bootstrap.<
...
 ---
 
 ```jsx live
 <>
   <Alert
     variant="info"
-    actions={[
-      <Button>Hello</Button>,
-    ]}
+    actions={[<Button>Hello</Button>]}
```
Comment (viktorrusakov) [inline src/Alert/README.md:68]: Can we not change this? The way it was before is more readable. Same in other places in this file

**Review by viktorrusakov — COMMENTED** [review]
@PKulkoRaccoonGang I left initial comments and will take a look later as well. 

I think you should change [printWidth](https://prettier.io/docs/en/options.html#print-width) setting in prettier config to match max line length setting that we have in [eslint config](https://github.com/openedx/eslint-config/blob/master/.eslintrc.js#L21). Otherwise, prettier does a lot of unnecessary changes by splitting what it thinks long lines into multiple lines

File: README.md (hunk 4 above)
Comment (PKulkoRaccoonGang) [reply README.md:339]: It seems to me that it is expected that the code that fits on one line will have this format

File: component-generator/templates/README.md (hunk 5 above)
Comment (PKulkoRaccoonGang) [reply component-generator/templates/README.md:18]: It seems to me that it is expected that the code that fits on one line will have this format

File: README.md (hunk 3 above)
Comment (PKulkoRaccoonGang) [reply README.md:197]: I think there is still a slight difference in indentation between these two designs

**Review by viktorrusakov — APPROVED** [review]

**Pull request merged**